streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -121,7 +123,7 @@ public void register(final StateStore store,
         }
 
         // check that the underlying change log topic exist or not
-        final String topic = storeToChangelogTopic.get(storeName);
+        final String topic = storeToChangelogTopic.get(storeName) == null ? stateStoreToChangelogTopicOnlyForRestoring.get(storeName) : storeToChangelogTopic.get(storeName);
```

Comment:
If storeToChangelogTopic.get(storeName) is not null, assert that stateStoreToChangelogTopicOnlyForRestoring doesn't contain storeName.